impeller/typographer/backends/stb/typeface_stb.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/typographer/backends/stb/typeface_stb.h"
+
+#include <cstring>
+
+#include "flutter/fml/logging.h"
+
+namespace impeller {
+
+TypefaceSTB::~TypefaceSTB() = default;
+
+// Instantiate a typeface based on a .ttf or other font file
+TypefaceSTB::TypefaceSTB(std::unique_ptr<fml::Mapping> typeface_mapping)
+    : typeface_mapping_(std::move(typeface_mapping)),
+      font_info_(std::make_unique<stbtt_fontinfo>()),
+      is_valid_(false) {
+  // We need an "offset" into the ttf file
+  auto offset = stbtt_GetFontOffsetForIndex(typeface_mapping_->GetMapping(), 0);
+  if (stbtt_InitFont(font_info_.get(), typeface_mapping_->GetMapping(),
```

Comment:
Is there a de-init font? Is this a leak?

Reply:
Nah, it just initializes the stbtt_fontinfo (first param).

Docstring:
```
// Given an offset into the file that defines a font, this function builds
// the necessary cached info for the rest of the system. You must allocate
// the stbtt_fontinfo yourself, and stbtt_InitFont will fill it out. You don't
// need to do anything special to free it, because the contents are pure
// value data with no additional data structures. Returns 0 on failure.
```